Warehouse.Management.Tests/EntryServiceTests.cs

```diff
@@ -0,0 +1,320 @@
+        mockUserService.Setup(x => x.UserId).Returns("TestUser");
+
+        var options = new DbContextOptionsBuilder<WarehouseManagementDbContext>()
+            .UseInMemoryDatabase(
+                databaseName: "WarehouseManagementTestDb" + Guid.NewGuid().ToString()
+            )
+            .Options;
+
+        dbContext = new WarehouseManagementDbContext(options, mockUserService.Object);
+
+        zone1 = new Zone
+        {
+            Id = 1,
+            Name = "Zone1",
+            CreatedByUserId = "User1"
+        };
+
+        zone2 = new Zone
+        {
+            Id = 2,
+            Name = "Zone2",
+            CreatedByUserId = "User1"
+        };
+
+        waitingEntry = new Entry
+        {
+            Id = 1,
+            Pallets = 0,
+            Packages = 6,
+            Pieces = 0,
+            Zone = zone1,
+            StartedProccessing = null,
+            FinishedProccessing = null,
+            CreatedByUserId = "User1"
+        };
+
+        processingEntry = new Entry
+        {
+            Id = 2,
+            Pallets = 4,
+            Packages = 0,
+            Pieces = 0,
+            Zone = zone1,
+            StartedProccessing = DateTime.UtcNow,
+            FinishedProccessing = null,
+            CreatedByUserId = "User1"
+        };
+
+        finishedEntry = new Entry
+        {
+            Id = 3,
+            Pallets = 0,
+            Packages = 0,
+            Pieces = 8,
+            Zone = zone2,
+            StartedProccessing = DateTime.UtcNow.AddDays(-7),
+            FinishedProccessing = DateTime.UtcNow,
+            CreatedByUserId = "User1"
+        };
+
+        deletedEntry = new Entry
+        {
+            Id = 4,
+            Pallets = 0,
+            Packages = 0,
+            Pieces = 5,
+            Zone = zone1,
+            StartedProccessing = null,
+            FinishedProccessing = null,
+            CreatedByUserId = "User1",
+            IsDeleted = true
+        };
+
+        var entries = new List<Entry> { waitingEntry, processingEntry, finishedEntry, deletedEntry };
+        var zones = new List<Zone> { zone1, zone2 };
+
+        await dbContext.Entries.AddRangeAsync(entries);
+        await dbContext.Zones.AddRangeAsync(zones);
+
+        await dbContext.SaveChangesAsync();
+
+        entryService = new EntryService(new Repository(dbContext, mockUserService.Object));
+    }
+
+    [TearDown]
+    public async Task Teardown()
+    {
+        await dbContext.Database.EnsureDeletedAsync();
+        await dbContext.DisposeAsync();
+    }
+
+    [Test]
+    public async Task CreateAsync_SuccessfullyCreatesEntry()
+    {
+        await entryService.CreateAsync(new EntryFormDto
+        {
+            Pallets = 0,
+            Packages = 0,
+            Pieces = 15
+        }, mockUserService.Object.UserId);
+
+        var addedEntry = await dbContext.Entries.FirstOrDefaultAsync(e => e.Pieces == 15);
+
+        Assert.IsNotNull(addedEntry);
+    }
```

Comment:
This check is now inside @PEAtanasov PR #30, but it would be nice to test when we try to add an entry with more than 1 type so maybe add you can prepare the test (make sure its failing) and just skip it to not run and when #30 is merged it will be enabled back again.

Reply:
I see what you mean, but the problem is that in https://github.com/InterCarsBulgaria-TechLeadAcademy/warehouse.management.be/pull/30, the `CreateAsync()` method is modified to accept a collection and after the merge of https://github.com/InterCarsBulgaria-TechLeadAcademy/warehouse.management.be/pull/30 the test will not be valid and will need refactoring.